packages/autocomplete-plugin-recent-searches/src/createStore.ts

```diff
@@ -0,0 +1,30 @@
+import { RecentSearchesItem } from './types/RecentSearchesItem';
+
+export type RecentSearchesStore<TItem extends RecentSearchesItem> = {
+  add(item: TItem): void;
+  remove(id: string): void;
+  getAll(query?: string): TItem[];
+};
+
+type CreateRecentSearchesStoreParams<TItem extends RecentSearchesItem> = {
+  onAdd(item: TItem): void;
+  onRemove(id: string): void;
+  getAll(query?: string): TItem[];
```

Comment:
Can it return a `Promise<TItem[]>`?
If customer uses their own server as a storage, we need to cover promises.

Reply:
Good call!

Reply:
See 370abaece212322d37a6954f30a3a52cbeb00ebb.